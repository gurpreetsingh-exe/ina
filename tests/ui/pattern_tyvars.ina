// COMPILE_FAIL

extern mod core;
using core::prelude::*;

using Foo::*;
type Foo =
    | T1(Option[isize])
    | T2

fn main() {
    match T2 {
        T1(Some[i32](a)) => {},
        T1(None) => {},
        T2 => {}
    }
}
