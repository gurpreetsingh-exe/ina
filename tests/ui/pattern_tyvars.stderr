error: mismatch types
    ╭─[tests/ui/pattern_tyvars.ina:13:12]
    │
 13 │         T1(Some[i32](a)) => {},
    │            ────┬────
    │                ╰───── expected `isize`, found `i32`
    ∙
────╯
